bin/faucet/src/client.rs

```diff
@@ -0,0 +1,353 @@
+
+    fn get_account_code(&self, account_id: AccountId) -> Result<ModuleAst, DataStoreError> {
+        let account = self.faucet_account.borrow();
+        if account_id != account.id() {
+            return Err(DataStoreError::AccountNotFound(account_id));
+        }
+
+        let module_ast = account.code().module().clone();
+        Ok(module_ast)
+    }
+}
+
+struct FaucetAuthenticator {
+    secret_key: AuthSecretKey,
+    rng: RefCell<ChaCha20Rng>,
+}
+
+impl FaucetAuthenticator {
+    pub fn new(secret_key: SecretKey) -> Self {
+        Self {
+            secret_key: AuthSecretKey::RpoFalcon512(secret_key),
+            rng: RefCell::new(ChaCha20Rng::from_entropy()),
+        }
+    }
+}
+
+impl TransactionAuthenticator for FaucetAuthenticator {
+    fn get_signature(
+        &self,
+        _pub_key: Word,
+        message: Word,
+        _account_delta: &AccountDelta,
+    ) -> Result<Vec<Felt>, AuthenticationError> {
+        let mut rng = self.rng.borrow_mut();
+        let AuthSecretKey::RpoFalcon512(k) = &self.secret_key;
+        get_falcon_signature(k, message, &mut rng)
+    }
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+fn build_account(config: FaucetConfig) -> Result<(Account, Word, SecretKey), FaucetError> {
+    let token_symbol = TokenSymbol::new(config.token_symbol.as_str())
+        .map_err(|err| FaucetError::AccountCreationError(err.to_string()))?;
+
+    let seed: [u8; 32] = [0; 32];
+
+    // Instantiate keypair and authscheme
+    let mut rng = ChaCha20Rng::from_seed(seed);
+    let secret = SecretKey::with_rng(&mut rng);
+    let auth_scheme = AuthScheme::RpoFalcon512 { pub_key: secret.public_key() };
+
+    let (faucet_account, account_seed) = create_basic_fungible_faucet(
+        seed,
+        token_symbol,
+        config.decimals,
+        Felt::try_from(config.max_supply)
+            .map_err(|err| FaucetError::InternalServerError(err.to_string()))?,
+        AccountStorageType::OffChain,
+        auth_scheme,
+    )
+    .unwrap();
+
+    Ok((faucet_account, account_seed, secret))
+}
+
+pub async fn initial_connection(
+    config: FaucetConfig,
+) -> Result<(ApiClient<Channel>, BlockHeader, ChainMmr), FaucetError> {
+    let endpoint = tonic::transport::Endpoint::try_from(config.node_url.clone())
+        .map_err(|_| FaucetError::InternalServerError("Failed to connect to node.".to_string()))?
+        .timeout(Duration::from_millis(config.timeout_ms));
+
+    let mut rpc_api = ApiClient::connect(endpoint)
+        .await
+        .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
+
+    let request = GetBlockHeaderByNumberRequest {
+        block_num: Some(0),
+        include_mmr_proof: Some(true),
+    };
+    let response = rpc_api.get_block_header_by_number(request).await.unwrap();
+    let root_block_header: BlockHeader =
+        response.into_inner().block_header.unwrap().try_into().unwrap();
```

Comment:
Perhaps instead of panicking here (and on line 296), the function should return a `FaucetError::InitializationError` or something similar, indicative of what the problem might be

Reply:
Done.